src/libraries/System.Net.Sockets/src/System/Net/Sockets/SafeSocketHandle.Windows.cs

```diff
@@ -59,6 +59,7 @@ internal ThreadPoolBoundHandle GetOrAllocateThreadPoolBoundHandle(bool trySkipCo
                     catch (Exception exception) when (!ExceptionCheck.IsFatal(exception))
                     {
                         bool closed = IsClosed;
+                        bool alreadyBound = !IsInvalid && !IsClosed && (exception is ArgumentException);
```

Comment:
Won't we get an ArgumentException as well if the handle wasn't opened for overlapped I/O?  We're assuming that's the case then?

Reply:
I made this based this on my understanding of the code in `ClrThreadPoolBoundHandle.Windows.cs`:

https://github.com/dotnet/runtime/blob/6ed4a5e3db0e0cefb6924881a99888a6e192a0f2/src/coreclr/src/System.Private.CoreLib/src/System/Threading/ClrThreadPoolBoundHandle.Windows.cs#L30-L34

I assume that the handle can not be invalid since we checked it before. Based on the comment, and the exception message, the only other possible reason is "Handle already bound or sync handle".  Let me know, if I'm missing something!
(The docs for win32 `BindIoCompletionCallback`  were not really helpful to me.)